hmda/src/main/scala/hmda/validation/dsl/PredicateHmda.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.dsl
+
+import java.text.SimpleDateFormat
+
+object PredicateHmda {
+
+  def validDateFormat[T]: Predicate[T] = (_: T) match {
+    case s: String =>
+      checkDateFormat(s)
+    case _ => false
+  }
+
+  private def checkDateFormat[T](s: String): Boolean = {
+    try {
+      val format = new SimpleDateFormat("yyyyMMdd")
```

Comment:
We had an issue using this date formatter a couple months ago, and I think we might run into it again here.

Check [this comment](https://github.com/cfpb/hmda-platform/issues/1498#issuecomment-365266710) for more information.